test/data_base.cpp

```diff
@@ -1402,12 +1402,14 @@ BOOST_AUTO_TEST_CASE(data_base__reorganize__pop_and_push__success)
     store_block_transactions(instance, block1, 1);
     BOOST_REQUIRE_EQUAL(instance.candidate(block1), error::success);
     test_heights(instance, 1u, 0u);
-
     auto block2 = read_block(MAINNET_BLOCK2);
     auto block2_header = block2.header();
     block2_header.set_previous_block_hash(genesis.hash());
     block2.set_header(block2_header);
-    const auto& block3 = read_block(MAINNET_BLOCK3);
+    auto block3 = read_block(MAINNET_BLOCK3);
+    auto block3_header = block3.header();
+    block3_header.set_previous_block_hash(block2.hash());
+    block3.set_header(block3_header);
```

Comment:
Why is reading block3 not sufficient to establish block2 as parent hash of block3?

Reply:
block2's previous_block_hash is changed to genesis.hash() and that results in block2's hash changing too. Therefore we need to update the block3's previous_block_hash.

Reply:
Okay, I see now. This is a very complex way to present header test branch. This doesn't give control over the work presented by the hash of the modified headers, which may result in unintended behavior.